crates/evm/src/evm/executor.rs

```diff
@@ -111,34 +106,25 @@ pub(crate) fn execute_multiple_tx<
             && *blob_gas_used + tx.blob_gas_used().unwrap() > MAX_DATA_GAS_PER_BLOCK
         {
             native_error!("Blob gas used exceeds block gas limit");
-            tx_results.push(Err(EVMError::Custom(format!(
-                "Blob gas used exceeds block gas limit {:?}",
-                block_gas_limit
-            ))));
-            continue;
+            return Err(SoftConfirmationModuleCallError::EvmBlobGasUsedExceedsBlockGasLimit);
         }
 
-        let result_and_state = match evm.transact(tx) {
-            Ok(result_and_state) => result_and_state,
-            Err(e) => {
-                native_error!(error = %e, "Transaction failed");
-                tx_results.push(Err(e));
-                continue;
-            }
-        };
+        let result_and_state = evm.transact(tx).map_err(|e| match e {
+            // only custom error we use is for not enough funds for L1 fee
+            EVMError::Custom(_) => SoftConfirmationModuleCallError::EvmNotEnoughFundsForL1Fee,
```

Comment:
What happens if we have multiple custom errors?

Reply:
then we'll figure something out :)